### DISCUSSION TITLE
PDF conversion

### DISCUSSION THREAD

**Discussion by mickaelperrin** [Tyrrrz/DiscordChatExporter#931] [General]
Hi,

Has someone successfully used a cli tool to do a proper HTML to PDF conversion ?

I tried several tools. 

wkhtmltopdf is the one which gives the best results but the presentation is still broken in some ways.

Maybe i miss some specific command line parameters or use a wrong version.

If someone has a linux/docker compatible conversion command, il would be great to share.

-- 

Some exports I used:

```
in docker:
docker run --rm -it -v $(pwd):/mount surnet/alpine-wkhtmltopdf:3.16.0-0.12.6-full --enable-local-file-access /mount/file.html /mount/file.pdf


native linux: (best result)
wkhtmltopdf 0.12.5
wkhtmltopdf --enable-local-file-access ./file.html ./file.pdf
```

**Comment by Technetium1** [comment]
@mickaelperrin https://github.com/jgm/pandoc

**Reply by mickaelperrin** [reply]
I give it a try, but it fails as soon as a ressource is not available or is not the proper type.

For example, I have an export referencing an image from imgur and the local file is an HTMLfile not an image one.

The conversion fail and I didn't found a way to continue even if an error occured.

```
docker run --rm --volume "`pwd`:/data" --user `id -u`:`id -g` pandoc/latex -s ./input.html -o ./output.pdf
[WARNING] Could not convert image /tmp/tex2pdf.-09e83db239d467fa/output_Files/CJMZPQI-EC1AB: Cannot load file
  Jpeg Invalid marker used
  PNG Invalid PNG file, signature broken
  Bitmap Invalid Bitmap magic identifier
  GIF Invalid Gif signature : <!doct
  HDR Invalid radiance file signature
  Tiff Invalid endian tag value
  TGA not enough bytes
Error producing PDF.
```
